src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -44,23 +51,39 @@ public JSONArray getSupportedLocalesFromRemoteVIP() {
 
     public List<Map> getAllComponentTranslation() {
         List<Map> list = new ArrayList<Map>();
-        Object[] locales = {};
-        Object[] components = {};
-        if (VIPCfg.getInstance().getMessageOrigin() == DataSourceEnum.VIP) {
-            locales = this.getSupportedLocalesFromRemoteVIP().toArray();
-            components = this.getComponentsFromRemoteVIP()
-                    .toArray();
-        }
-        for (Object locale : locales) {
-            for (Object component : components) {
-                dto.setComponent(((String) component).trim());
-                dto.setLocale(LocaleUtility.fmtToMappedLocale((String) locale).toString().trim());
-                Map<String, String> retMap = new ComponentService(dto).getMessages().getCachedData();
-                if (retMap != null) {
-                    list.add(retMap);
+        Map<String, String> locales = this.getLanguages(VIPCfg.getInstance().getMsgOriginsQueue().iterator());
+        List<String> components = this.getComponents(VIPCfg.getInstance().getMsgOriginsQueue().iterator());
+        if (locales != null) {
+            for (String languageTag : locales.keySet()) {
+                for (Object component : components) {
+                    dto.setComponent(((String) component).trim());
+                    dto.setLocale(LocaleUtility.fmtToMappedLocale(Locale.forLanguageTag(languageTag)).toString().trim());
+                    Map<String, String> retMap = new ComponentService(dto).getMessages().getCachedData();
+                    if (retMap != null) {
+                        list.add(retMap);
+                    }
                 }
             }
         }
         return list;
     }
+
+    private Map<String, String> getLanguages(Iterator<DataSourceEnum> msgSourceQueueIter) {
+        if (!msgSourceQueueIter.hasNext())
+            return null;
+
+        DataSourceEnum dataSource = msgSourceQueueIter.next();
+        LocaleOpt opt = dataSource.createLocaleOpt();
+        return opt.getLanguages(LocaleUtility.getDefaultLocale().toLanguageTag());
```

Comment:
If the map is empty, still need to get from the next iter in the queue.

Reply:
Fixed.